packages/slate/src/interfaces/custom-types.ts

```diff
@@ -0,0 +1,18 @@
+import { Descendant } from './node'
+
+export interface BaseElement {
```

Comment:
I think I would move `BaseElement` next to the `Element` type and `BaseText` next to the `Text` type (i.e., back in the interface files for those types).  Easier to read/follow the source that way.